langserve/server.py

```diff
@@ -691,6 +705,17 @@ async def config_schema_with_config(
             dependencies=dependencies,
         )(api_handler._check_feedback_enabled)
 
+    if enable_public_trace_link_endpoint:
+        app.put(
+            namespace + "/public_trace_link",
+            dependencies=dependencies,
+        )(api_handler.create_public_trace_link)
+
+        app.head(
+            namespace + "/public_trace_link",
+            dependencies=dependencies,
+        )(api_handler._check_public_trace_link_enabled)
```

Comment:
could we avoid using the private one. 

the client should be getting a true/false response rather than checking http status codes to check if it's enabled.

we can keep it as head or change to get endpoint and different name no preference on that part from me

Reply:
Discussed - will leave for now and deprecate later